critiquebrainz/data/dump_manager.py

```diff
@@ -269,47 +321,68 @@ def importer(archive):
+        import_data(os.path.join(temp_dir, 'cbdump', 'license'), 'license')
+        import_data(os.path.join(temp_dir, 'cbdump', 'review'), 'review')
+        import_data(os.path.join(temp_dir, 'cbdump', 'revision'), 'revision')
 
         shutil.rmtree(temp_dir)  # Cleanup
         print("Done!")
 
 
-def import_data(file_name, model, columns=None):
-    db_connection = db.session.connection().connection
-    cursor = db_connection.cursor()
-    try:
-        with open(file_name) as f:
-            # Checking if table already contains any data
-            if model.query.count() > 0:
-                print("Table %s already contains data. Skipping." % model.__tablename__)
-                return
-            # and if it doesn't, trying to import data
-            print("Importing data into %s table." % model.__tablename__)
-            if columns is None:
-                columns = get_columns(model)
-            cursor.copy_from(f, '"%s"' % model.__tablename__, columns=columns)
-            db_connection.commit()
-    except IOError as exception:
-        if exception.errno == errno.ENOENT:
-            print("Can't find data file for %s table. Skipping." % model.__tablename__)
-        else:
-            sys.exit("Failed to open data file. Error: %s" % exception)
+def import_data(file_name, tablename, columns=None):
+
+    if tablename == "user":
+        tablename = '"user"'
+
+    # Explode database uri
+    db_hostname, db_port, db_name, db_username, db_password = explode_db_uri(current_app.config['SQLALCHEMY_DATABASE_URI'])
+
+    # Checking if table already contains any data
+    with db.engine.connect() as connection:
+        count = connection.execute(sqlalchemy.text("""
+            SELECT COUNT(*)
+              FROM {tablename}
+        """.format(tablename=tablename)))
+        count = count.fetchone()[0]
+    if count > 0:
+        print("Table %s already contains data. Skipping." % tablename)
+        return
+    # Checking if the specified file exists or if the file is empty
+    elif not os.path.exists(file_name) or os.stat(file_name).st_size == 0:
+            print("Can't find data file for %s table. Skipping." % tablename)
+    # and if it doesn't, trying to import data
+    else:
+        print("Importing data into %s table." % tablename)
+        try:
+            if columns:
+                result = subprocess.call("""
+                    psql -h {} -p {} -U {} -d {} -a -c '\COPY {tablename}({columns}) FROM {file_name}'
+                """.format(db_hostname, db_port, db_username, db_password, tablename=tablename,
+                        columns=", ".join(list(columns)), file_name=file_name), shell=True)
+                if result != 0:
+                    raise Exception("Failed to copy data for table %s" % tablename)
+            else:
+                result = subprocess.call("""
+                    psql -h {} -p {} -U {} -d {} -a -c '\COPY {tablename} FROM {file_name}'
```

Comment:
Yes, it does, if data for all the columns of a table is present in the file and columns are in the same order as the table structure. The `public` function for making db dumps creates files ensuring the same order of columns.